### DISCUSSION TITLE
Unable to test unavoidable manually edited migration against original seed data - because it runs AFTER all migrations

### DISCUSSION THREAD

**Discussion by hardysabs2** [prisma/prisma#21509] [Q&A]
### Question

If your data is already in production on occasion one must adapt a complex migration to include defaults and data manipulation, but because the seed must run after all the migrations it is not possible to always test it.

For example
Branch off dev to feature branch
set empty feature db
npx prisma migrate dev (includes seed) 
schema change (this change includes a utility table that requires data inserted and on a table which already has non-removable data, a change of type and a default value on a field that will become child FK to the utility table)
npx prisma migrate dev --create-only
edit the migration manually to add utility table data and position the 'new FK with default' change after parent utility table is populated
prisma migrate dev
now we have a migration run that has been tested against data of the form it will run against in production
so can successfully run in production
however the original seed data is effectively broken because when next run it is forced to run AFTER this edited migration
and this edited migration can never be properly tested with data again because next time it will run when the database is empty

While we are waiting for more flexibility for prisma how do we manage situations like this?
Just like we have done? and fix the seed data to match the state AFTER this production suitable migration, but never properly test the migration again as in dev it will always run on an empty child table?
Or some better way?

If there were a way to run the seed only after the migration that represents the point of production release all the subsequent migrations that have no option of tear down can be tested to successfully run on production data.

Thanks,

Andrew

**Comment by ludralph** [comment]
Hi @hardysabs2 👋 

Currently, there is no way to test migrations with seed data at specific points in the migration history. We do have two feature request [#18059](https://github.com/prisma/prisma/issues/18059) and [#6345](https://github.com/prisma/prisma/issues/6345) that looks relevant for your usecase.

For the time being, you might need to continue with your current approach of manually adjusting the seed data to match the state after the production-suitable migration.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
